src/server.c

```diff
@@ -622,13 +622,15 @@ int incrementallyRehash(int dbid) {
  * as we want to avoid resizing the hash tables when there is a child in order
  * to play well with copy-on-write (otherwise when a resize happens lots of
  * memory pages are copied). The goal of this function is to update the ability
- * for dict.c to resize the hash tables accordingly to the fact we have an
+ * for dict.c to resize or rehash the tables accordingly to the fact we have an
  * active fork child running. */
-void updateDictResizePolicy(void) {
-    if (!hasActiveChildProcess())
-        dictEnableResize();
+void updateDictResizePolicy() {
+    if (server.in_fork_child != CHILD_TYPE_NONE)
+        dictSetResizeEnabled(DICT_RESIZE_FORBID);
```

Comment:
Can it ever happen that a child process wants to use a dict for internal bookkeeping of some sort? A dict that the child has created and which is not shared with the parent process. Since enable-resize is global, FORBID would prevent that use case.

Would it be enough to use AVOID in the child process? Maybe it solves the problem just as well in practice. If the child isn't adding anything to the CoW-shared dicts, they will never reach force-resize...

Reply:
using AVOID here means that `dict_can_resize` becomes boolean again, and we're back with this issue:
https://github.com/redis/redis/pull/11692#discussion_r1063157357
that is unless we want to allow the child to do gradual rehashing in parallel to what the parent does.
i.e. the parent started a rehashing to x8, it continues to do that during the fork, and the child does that too.
note that there aren't many calls to dictFind/add/del in the child, the one i saw was from getExpire.

anyway, i don't think i wanna handle that issue, if we'll one day need to create temporary dicts in the child, we'll have to either pre-allocate them to the right size, or add some per-dict manual override that will ignore this global.

Reply:
OK, sounds good. Keep FORBID.

> using AVOID here means that dict_can_resize becomes boolean again, and we're back with this issue

That's true, but in practice without this PR, IIUC it is ALLOW in the child, right? With AVOID the rehashing is at least less likely to happen.